lance-spark-base_2.12/src/test/java/com/lancedb/lance/spark/write/VectorWriteTest.java

```diff
@@ -0,0 +1,234 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.lancedb.lance.spark.write;
+
+import com.lancedb.lance.spark.LanceConfig;
+import com.lancedb.lance.spark.LanceDataSource;
+
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.MetadataBuilder;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class VectorWriteTest {
+  private static SparkSession spark;
+
+  @BeforeAll
+  static void setup() {
+    spark =
+        SparkSession.builder()
+            .appName("vector-write-test")
+            .master("local[*]")
+            .config("spark.sql.execution.arrow.maxRecordsPerBatch", "1024")
+            .config("spark.sql.catalog.lance", "com.lancedb.lance.spark.LanceCatalog")
+            .getOrCreate();
+  }
+
+  @AfterAll
+  static void tearDown() {
+    if (spark != null) {
+      spark.stop();
+    }
+  }
+
+  @Test
+  public void testVectorColumnFloat32(TestInfo testInfo) {
+    int numRows = 256;
+    int vectorDim = 128;
+
+    // Create metadata for vector column
+    Metadata vectorMetadata =
+        new MetadataBuilder().putLong("arrow.FixedSizeList.size", vectorDim).build();
+
+    // Create schema with vector column
+    StructType schema =
+        new StructType(
+            new StructField[] {
+              DataTypes.createStructField("id", DataTypes.IntegerType, false),
+              DataTypes.createStructField(
+                  "embeddings",
+                  DataTypes.createArrayType(DataTypes.FloatType, false),
+                  false,
+                  vectorMetadata)
+            });
+
+    // Create test data
+    List<Row> rows = new ArrayList<>();
+    Random random = new Random(42);
+    for (int i = 0; i < numRows; i++) {
+      float[] vector = new float[vectorDim];
+      for (int j = 0; j < vectorDim; j++) {
+        vector[j] = random.nextFloat();
+      }
+      rows.add(RowFactory.create(i, vector));
+    }
+
+    Dataset<Row> df = spark.createDataFrame(rows, schema);
+
+    // Write to Lance format
+    String outputPath = "/tmp/vector_test_float32.lance";
+    deleteDirectory(new File(outputPath));
+
+    df.write()
```

Comment:
this probably covers the data frame write case. What about if we just run CREATE TABLE to create an empty table? for example https://github.com/lancedb/lance-spark/blob/main/lance-spark-base_2.12/src/test/java/com/lancedb/lance/spark/SparkLanceNamespaceTestBase.java#L135.

I think we can do something like

```
CREATE TABLE (vector ARRAY<FLOAT>)
TBLPROPERTIES (
  'vector.arrow.fixed-size-list.size'='1024'
)
```

And then you can do the conversion at https://github.com/lancedb/lance-spark/blob/main/lance-spark-base_2.12/src/main/java/com/lancedb/lance/spark/utils/SchemaConverter.java to take the additional table properties values at https://github.com/lancedb/lance-spark/blob/main/lance-spark-base_2.12/src/main/java/com/lancedb/lance/spark/LanceNamespaceSparkCatalog.java#L408

Reply:
Good call! Forget about SQL statement!, have updated to use your approach